modules/videoio/src/cap_dshow.cpp

```diff
@@ -101,7 +101,12 @@ Thanks to:
 
 #include <tchar.h>
 #include <stdlib.h>
+#if !defined(__MINGW32__) && !defined(__MINGW64__)
+// MinGW sprintf is would be defined as non stdio one. So 
+// include this later after dshow.h declaration, otherwise compile failure 
+// because mingw dshow sprintf has it's own declaration that is different from the standard
 #include <stdio.h>
```

Comment:
Mentioned header has been included by precomp.hpp before that. So this condition looks useless.

Please provide:
- error message from compiler without patch
- used version of MinGW installation (with link)